### DISCUSSION TITLE
Matches returned limit

### DISCUSSION THREAD

**Discussion by Cimlah** [Henrik-3/unofficial-valorant-api#65] [Q&A] [answered]
Hello 👋,
I'd like to ask a question about limitation of matches returned by `/valorant/v1/by-puuid/lifetime/matches/{affinity}/{puuid}` endpoint. Why does this endpoint return only 10 latest matches? I'm building a Discord bot for my private server, which will display total KDA and for this I need access to every match in order to sum kills, deaths and assists. **Is there any way to increase the limit?** Or could I perhaps host this API on my own and manage limits by myself on my server?
Thanks in advance for answers.

Have a good day
Michał

**Comment by raimannma** [comment, marked as answer]
There is no limit, the API returns all matches that are cached on API server.

If you keep fetching this endpoint repeatedly, you will see more and more matches.

This API is currently closed source. You can use the in-game API directly tough if you like to.

**Reply by Cimlah** [reply]
Okay, thanks. Perhaps there is some problem with caching and/or Riot servers have a delay.

**Comment by raimannma** [comment]
Riot Servers do not send all matches, they just don't provide that functionality.
Caching will just cache all matches that got requested from this API, so if you never requested any match for a user they won't be cached.


-------- Ursprüngliche Nachricht --------
Von: Michał Stolarczyk ***@***.***>
Datum: 09.07.23 13:36 (GMT+01:00)
An: Henrik-3/unofficial-valorant-api ***@***.***>
Cc: Manuel Raimann ***@***.***>, Comment ***@***.***>
Betreff: Re: [Henrik-3/unofficial-valorant-api] Matches returned limit (Discussion #65)


Okay, thanks. Perhaps there is some problem with caching and/or Riot servers have a delay.
